ocrd/ocrd/resource_manager.py

```diff
@@ -13,27 +13,46 @@
 from ocrd_validators import OcrdResourceListValidator
 from ocrd_utils import getLogger
 from ocrd_utils.os import list_all_resources, pushd_popd
-from ocrd_utils.constants import XDG_CONFIG_HOME, XDG_DATA_HOME
-
 from .constants import RESOURCE_LIST_FILENAME, RESOURCE_USER_LIST_COMMENT
 
 class OcrdResourceManager():
 
     """
     Managing processor resources
     """
-    def __init__(self):
+    def __init__(self, xdb_config_home=None, xdb_data_home=None):
         self.log = getLogger('ocrd.resource_manager')
         self.database = {}
+
+        self.xdb_data_home = xdb_data_home
+        if not self.xdb_data_home:
+            self._init_data_home()
```

Comment:
This could probably be solved in a property:

```
@property
def xdg_home(self):
  if not gettattr(self, '_xdg_home', None):
        the_home = path.expanduser('~')
        if 'HOME' in environ and environ['HOME'] != path.expanduser('~'):
            the_home = environ['HOME']
        self._xdg_home = the_home
  return self._xdg_home
```

etc.

Reply:
If you like to, it'll we done this way

Reply:
If I can choose, then yes please :) I really like `@property`, they combine the advantages of flat attributes with  methods and are great for encapsulating initialization.

Reply:
Turned `xdg_data_home` into a property, but not `xdg_config_home` since this information is already required by the resource_manager at initialization time for user_list.

Reply:
OK, but why not use the property in the constructor? For consistency I added `userdir` (your `the_home` variable) and `xdg_config_home` to the constructor kwargs. Spotted a bug already (database not reloaded after adding to it) - hooray for testing :)